ebb_lint/flake8.py

```diff
@@ -247,8 +252,18 @@ def _message_for_pos(self, pos, error, **kw):
         return lineno, column, message, type(self)
 
     def run(self):
-        with open(self.filename, 'r') as infile:
-            self.future_features = detect_future_features(infile)
+        if self.filename == 'stdin':
+            try:
+                enc = getattr(sys.stdin, 'encoding') or 'utf-8'
+                stdin_reader = codecs.getreader(enc)(sys.stdin)
+                self.future_features = detect_future_features(stdin_reader)
+            finally:
+                # like the `with open()` pattern, tries to ensure we can read
+                # from this source again in the future.
+                sys.stdin.seek(0)
+        else:
+            with open(self.filename, 'r') as infile:
+                self.future_features = detect_future_features(infile)
```

Comment:
Yup! :( Fortunately this is really easy to reproduce, per the one commit message—and even if it's kinda messy, at least it's an easy condition to branch on.